contracts/AirdropRegistry.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.15;
+
+/// @title AirdropRegsitry - Contract that holds state of all AirbroCampaignFactory contracts and their iterations
+contract AirdropRegistry {
+    address public immutable treasury;
+    // index of deployed airdrop contracts
+    address[] public airdrops;
+    address public admin;
+
+    uint256 public totalAirdropsCount;
+    // list of whitelisted and blacklisted AirbroCampaignFactory contract addresses
+    mapping(address => bool) public factories;
+
+    event FactoryWhitelisted(address indexed factoryAddress);
+    event FactoryBlacklisted(address indexed factoryAddress);
+    event NewAirdrop(address indexed _airdropContract, address indexed _creator, string indexed _airdropType);
+    event AdminChanged(address indexed adminAddress);
+
+    error NotAdmin();
+    error NotWhitelisted();
+
+    modifier onlyAdmin() {
```

Comment:
Every contract pretty much follows the same pattern for admin, I understand that you don't want to use OZ ownable, because of the custom errors. But you can make your own ownable and inherit in child contracts without too much repetition

Reply:
aha, you mean we could make like a shared contract that includes the admin var, onlyAdmin modifier and changeAdmin setter function and then inherit it instead of typing it everywhere @bxmmm1

Reply:
Yes